content/master/concepts/patch-and-transform.md

```diff
@@ -0,0 +1,1692 @@
+them to the managed resource.
+
+{{<hint "tip" >}}
+Use 
+{{<hover label="fromEnvField" line="12">}}FromEnvironmentFieldPath{{</hover>}}
+to apply custom managed resource settings based on the current environment.  
+{{< /hint >}}
+
+For example, use the environment's 
+{{<hover label="fromEnvField" line="13">}}locations.eu{{</hover>}} value and
+apply it as the 
+{{<hover label="fromEnvField" line="14">}}region{{</hover>}}.
+
+```yaml {label="fromEnvField",copy-lines="9-11"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for brevity
+    - name: bucket1
+      base:
+        apiVersion: s3.aws.upbound.io/v1beta1
+        kind: Bucket
+        spec:
+          forProvider:
+            region: us-east-2
+        patches:
+        - type: FromEnvironmentFieldPath
+          fromFieldPath: locations.eu
+          toFieldPath: spec.forProvider.region
+```
+
+Verify managed resource to confirm the applied patch. 
+
+```yaml {copy-lines="none"}
+kubectl describe bucket
+Name:         my-example-claim-8vrvc-xx5sr
+Labels:       crossplane.io/claim-name=my-example-claim
+# Removed for brevity
+Spec:
+  For Provider:
+    Region:  eu-north-1
+  # Removed for brevity
+```
+
+<!-- vale Google.Headings = NO -->
+### ToEnvironmentFieldPath
+<!-- vale Google.Headings = YES -->
+
+{{<hint "important" >}}
+EnvironmentConfigs are an alpha feature. They aren't enabled by default.  
+
+For more information about using an EnvironmentConfig, read the [Environment
+Configs]({{<ref "./environment-configs">}}) documentation.
+{{< /hint >}}
+
+The 
+{{<hover label="toEnvField" line="12">}}ToEnvironmentFieldPath{{</hover>}}
+patch takes values the managed resource and applies them to the in-memory 
+EnvironmentConfig environment.
+
+{{<hint "tip" >}}
+Use 
+{{<hover label="toEnvField" line="12">}}ToEnvironmentFieldPath{{</hover>}}
+write data to the environment that any FromEnvironmentFieldPath
+patch can access. 
+{{< /hint >}}
+
+For example, use the desired
+{{<hover label="toEnvField" line="13">}}region{{</hover>}} value and
+apply it as the environment's
+{{<hover label="toEnvField" line="14">}}key1{{</hover>}}.
+
+{{< hint "important" >}}
+The environment's key must already exist. Patches can't create new environment
+keys. 
+{{< /hint >}}
+
+```yaml {label="toEnvField",copy-lines="9-11"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+# Removed for brevity
+    - name: bucket1
+      base:
+        apiVersion: s3.aws.upbound.io/v1beta1
+        kind: Bucket
+        spec:
+          forProvider:
+            region: us-east-2
+        patches:
+        - type: ToEnvironmentFieldPath
+          fromFieldPath: spec.forProvider.region
+          toFieldPath: key1
+```
+
+Because the environment is in-memory, there is no command to confirm the patch
+wrote the value to the environment.
+
+{{<hint "important">}}
+The 
+{{<hover label="toEnvField" line="12">}}ToEnvironmentFieldPath{{</hover>}}
```

Comment:
this is a bit surprising to me - did you get a sense of if this was intentional, or is it a bug in the behavior of `EnvironmentConfig`?

Reply:
It seem intentional? It's mentioned in the existing docs
https://github.com/crossplane/crossplane/issues/3926#issuecomment-1578886896

But I also think it's a poor user experience and drove me a little crazy on trying to figure out what can and can't be patched.